### PR TITLE
Moddcb 102

### PR DESCRIPTION
**Pull request by SreejaMangarapu**
## Purpose
https://folio-org.atlassian.net/browse/MODDCB-101 
https://folio-org.atlassian.net/browse/MODDCB-102

## Approach
Request Event:
added a condition in request event listener which checks whether the payload contain instanceTitle as DCB_INSTANCE or requesterLastName as DcbSystem.

Loan Event:
in mod circulation , if the event is related to dcb then a boolean flag isDcb = true is added to the payload , for non dcb events it is false , so in the loan event listener of mod-dcb added a condition which checks whether the event is dcb or non dcb event with the help of isdcb flag

#### TODOS and Open Questions
- [ ] Use GitHub checklists. When solved, check the box and explain the answer.

## Learning
_Describe the research stage. Add links to blog posts, patterns, libraries or addons used to solve this problem._

### CODE REVIEW

File: src/main/java/org/folio/dcb/listener/kafka/CirculationEventListener.java [hunk 1]
```diff
@@ -40,19 +40,20 @@ public class CirculationEventListener {
   public void handleLoanEvent(String data, MessageHeaders messageHeaders) {
     String tenantId = getHeaderValue(messageHeaders, XOkapiHeaders.TENANT, null).get(0);
     var eventData = parseLoanEvent(data);
-    if (Objects.nonNull(eventData)) {
+    if (Objects.nonNull(eventData) && eventData.isDcb()) {
```
Comment (gurleenkaurbp) [inline src/main/java/org/folio/dcb/listener/kafka/CirculationEventListener.java:43]: This condition should be changed and the case when isDcb flag is not present in  the payload the if condition should also pass. Please check

File: src/main/java/org/folio/dcb/listener/kafka/CirculationEventListener.java [hunk 2]
```diff
@@ -71,22 +72,27 @@ public void handleRequestEvent(String data, MessageHeaders messageHeaders) {
...
-                baseLibraryService.updateTransactionEntity(transactionEntity, TransactionStatus.StatusEnum.AWAITING_PICKUP);
-              } else {
-                log.info("handleRequestEvent:: status for event {} can not be updated", eventData.getType());
-              }
-            })
-        );
+      String instanceTitle = eventData.getInstanceTitle();
+      String requesterLastName = eventData.getRequesterLastName();
+      if ("DCB_INSTANCE".equals(instanceTitle) || "DcbSystem".equals(requesterLastName)) {
+        log.info("dcb flow for a request event");
```
Comment (gurleenkaurbp) [inline src/main/java/org/folio/dcb/listener/kafka/CirculationEventListener.java:78]: please make this as debug log

File: src/main/java/org/folio/dcb/listener/kafka/CirculationEventListener.java [hunk 3]
```diff
@@ -40,19 +40,20 @@ public class CirculationEventListener {
   public void handleLoanEvent(String data, MessageHeaders messageHeaders) {
     String tenantId = getHeaderValue(messageHeaders, XOkapiHeaders.TENANT, null).get(0);
     var eventData = parseLoanEvent(data);
-    if (Objects.nonNull(eventData)) {
+    if (Objects.nonNull(eventData) && eventData.isDcb()) {
+      log.info("dcb flow for a loan event");
```
Comment (gurleenkaurbp) [inline src/main/java/org/folio/dcb/listener/kafka/CirculationEventListener.java:44]: it should become a debug log type

File: src/main/java/org/folio/dcb/utils/TransactionHelper.java [hunk 4]
```diff
@@ -37,6 +37,9 @@ public static EventData parseLoanEvent(String eventPayload) {
             eventData.setType(EventData.EventType.CHECK_IN);
           }
         }
+        if (kafkaEvent.getNewNode().has("isDcb")) {
```
Comment (gurleenkaurbp) [inline src/main/java/org/folio/dcb/utils/TransactionHelper.java:40]: please use string constant for isDcb

File: src/main/java/org/folio/dcb/utils/TransactionHelper.java [hunk 5]
```diff
@@ -55,6 +58,13 @@ public static EventData parseRequestEvent(String eventPayload){
           case CLOSED_CANCELLED -> eventData.setType(EventData.EventType.CANCEL);
           default -> log.info("parseRequestEvent:: Request status {} is not supported", requestStatus);
         }
+        if (kafkaEvent.getNewNode().has("instance") && kafkaEvent.getNewNode().get("instance").has("title")) {
```
Comment (gurleenkaurbp) [inline src/main/java/org/folio/dcb/utils/TransactionHelper.java:61]: Use string constants for instance and title value

File: src/main/java/org/folio/dcb/utils/TransactionHelper.java [hunk 6]
```diff
@@ -55,6 +58,13 @@ public static EventData parseRequestEvent(String eventPayload){
...
+          eventData.setInstanceTitle(kafkaEvent.getNewNode().get("instance").get("title").asText());
+        }
+
+        if (kafkaEvent.getNewNode().has("requester") && kafkaEvent.getNewNode().get("requester").has("lastName")) {
+          eventData.setRequesterLastName(kafkaEvent.getNewNode().get("requester").get("lastName").asText());
+        }
```
Comment (gurleenkaurbp) [inline src/main/java/org/folio/dcb/utils/TransactionHelper.java:67]: string constants should be used

**Review by gurleenkaurbp — APPROVED** [review]

**Review by Vignesh-kalyanasundaram — APPROVED** [review]

**Pull request merged**